pkg/parse/utils.go

```diff
@@ -99,3 +99,51 @@ func primitiveFromNativeDataType(native antlr.TerminalNode) (*sysl.Type_Primitiv
 	}
 	return &sysl.Type_Primitive_{Primitive: primitiveType}, constraint
 }
+
+type PathStack struct {
+	sep   string
+	parts []string
```

Comment:
nit: I suspect the code will end up simpler without `parts`. Since most calls invoke `p.update()`, it might also be more efficient.

Reply:
`p.update()` was really just premature optimisation and should/could probably be removed.

I'm actually hoping to remove the need for this completely in follow up PR's.